s4e-web/src/app/views/settings/components/generic-list-view/generic-list-view.component.ts

```diff
@@ -0,0 +1,74 @@
+import {Component, ContentChild, EventEmitter, Input, OnDestroy, OnInit, Output, TemplateRef} from '@angular/core';
+import {FormArray, FormControl} from '@ng-stack/forms';
+import {untilDestroyed} from 'ngx-take-until-destroy';
+import {ID} from '@datorama/akita';
+import {Subscription} from 'rxjs';
+import {debounceTime} from 'rxjs/operators';
+
+@Component({
+  selector: 's4e-generic-list-view',
+  templateUrl: './generic-list-view.component.html',
+  styleUrls: ['./generic-list-view.component.scss']
+})
+export class GenericListViewComponent implements OnInit, OnDestroy {
+  @ContentChild(TemplateRef) tableRow: TemplateRef<any>;
+  @Input() error: any|null = null;
+  @Input() loading: boolean = true;
+  @Input() itemIdFunction: (any) => ID = (item: any) => item.id;
+  @Input() itemSearchFunction: (any, query) => boolean = (item: any, query: string) => Object.values(item)
+    .filter(v => typeof v == 'string').map((v: string) => v.toLocaleLowerCase()).join(';').indexOf(query.toLocaleLowerCase()) >= 0;
+
+  protected _originalItems: any[] = [];
+  items: any[] = [];
+  @Input('items') set _items(val: any[]) {
+    this._originalItems = val;
+    this.items = this._originalItems;
+
+    if(this.searchable && this.offlineSearch && this.queryFc.value.length > 0) {
+      this.items = this._originalItems.filter(item => this.itemSearchFunction(item, this.queryFc.value));
+    }
+
+    if(this.selectable) {
+      if (this._itemSelectSub) {
+        this._itemSelectSub.unsubscribe();
+      }
+
+      this.itemSelectFc = new FormArray(val.map(item => new FormControl<boolean>(false)));
+      this._itemSelectSub = this.itemSelectFc.valueChanges.subscribe(val => {
+        const selected = val.map((val, i) => val && this.itemIdFunction(this.items[i])).filter(v => !!v);
+        this.onselect.emit(selected);
+        this.selectAllFc.setValue(selected.length > 0, {emitEvent: false});
+      });
+    }
+  }
+
+  @Input() selectable: boolean = false;
+  @Input() offlineSearch: boolean = true;
+  @Input() searchable: boolean = false;
+  @Output() onquery = new EventEmitter<string>();
+  @Output() onselect = new EventEmitter<ID[]>();
+
+  queryFc: FormControl<string> = new FormControl<string>('');
+  selectAllFc: FormControl<boolean> = new FormControl<boolean>(false);
+  itemSelectFc: FormArray<boolean>;
+
+  private _itemSelectSub: Subscription = null;
+
+  constructor() { }
+
+  ngOnInit() {
+    this.selectAllFc.valueChanges.pipe(untilDestroyed(this)).subscribe(value => {
+      this.itemSelectFc.setValue(this.itemSelectFc.controls.map(() => value));
+    });
+
+    this.queryFc.valueChanges.pipe(debounceTime(250), untilDestroyed(this)).subscribe(value => {
+      if(this.offlineSearch) {
```

Comment:
Could you change the `value` name to more meaningfull, like `formValues`?

Reply:
this is value of the single form control, I think that this is pretty self explanatory...